assets/js/components/Table/Table.test.jsx

```diff
@@ -0,0 +1,113 @@
+import React from 'react';
+import { useSearchParams } from 'react-router-dom';
+
+import { screen, fireEvent } from '@testing-library/react';
+import 'intersection-observer';
+import '@testing-library/jest-dom';
+
+import { faker } from '@faker-js/faker';
+import { Factory } from 'fishery';
+
+import { renderWithRouter } from '@lib/test-utils';
+
+import { filterTable } from '@lib/test-utils/table';
+
+import Table from './Table';
+
+const tableConfig = {
+  pagination: true,
+  columns: [
+    {
+      title: 'Column1',
+      key: 'column1',
+      filter: true,
+      filterFromParams: true,
+    },
+    {
+      title: 'Column2',
+      key: 'column2',
+      filter: true,
+      filterFromParams: true,
+    },
+    {
+      title: 'Column3',
+      key: 'column3',
+      filter: true,
+      filterFromParams: true,
+    },
+  ],
+};
+
+const tableFactory = Factory.define(() => ({
+  column1: faker.name.firstName(),
+  column2: faker.name.firstName(),
+  column3: faker.name.firstName(),
+}));
+
+function TestTable({ config, data }) {
+  const [searchParams, setSearchParams] = useSearchParams();
+
+  return (
+    <Table
+      config={config}
+      data={data}
+      searchParams={searchParams}
+      setSearchParams={setSearchParams}
+    />
+  );
+}
+
+describe('Table component', () => {
+  describe('filtering', () => {
+    it('should filter by the chosen filter option', () => {
+      const data = tableFactory.buildList(10);
+      const { column1: value1 } = data[0];
+
+      renderWithRouter(<TestTable config={tableConfig} data={data} />);
+
+      filterTable('Column1', value1);
+
+      const table = screen.getByRole('table');
+      expect(table.querySelectorAll('tbody > tr')).toHaveLength(1);
+    });
+
+    it('should put the filters values in the query string when filters are selected', () => {
+      const data = tableFactory.buildList(10);
+      const { column1: value1, column2: value2, column3: value3 } = data[0];
+
+      renderWithRouter(<TestTable config={tableConfig} data={data} />);
+
+      ['Column1', 'Column2', 'Column3'].forEach((filter) => {
+        fireEvent.click(screen.getByTestId(`filter-${filter}`));
+
+        fireEvent.click(
+          screen
+            .getByTestId(`filter-${filter}-options`)
+            .querySelector('li > div > span').firstChild
+        );
+      });
+
+      expect(window.location.search).toEqual(
+        `?column1=${value1}&column2=${value2}&column3=${value3}`
+      );
+    });
```

Comment:
You could test this side effect in a lot of different ways. This is the first, but it requires React Router, and we aren't really focusing on the filtering behavior but more on the interaction with the browser.

We could also test that for example filling a list with filters behaves the correct way, using plain functions instead of the ones that come from React Router. What do you think?

Reply:
I will add these tests back to the overview tests.
I had it there at the beginning, but they looked exactly the same, that's why I put them here.
But i'm having moving them back